src/sidebar/services/threads.js

```diff
@@ -1,15 +1,24 @@
+/**
+ * @typedef {import('../util/build-thread').Thread} Thread
+ */
+
 // @ngInject
 export default function threadsService(store) {
   /**
    * Make this thread and all of its children "visible". This has the effect of
    * "unhiding" a thread which is currently hidden by an applied search filter
-   * (as well as its child threads).
+   * (as well as its child threads). Only threads that are not currently visible
+   * will be forced visible.
+   *
+   * @param {Thread} thread
    */
   function forceVisible(thread) {
     thread.children.forEach(child => {
       forceVisible(child);
     });
-    store.setForcedVisible(thread.id, true);
+    if (!thread.visible) {
+      store.setForcedVisible(thread.id, true);
```

Comment:
Is this just an optimization?

Reply:
Nope! It makes calculating the actual filter/search result count versus the forced-expanded count much easier.

In the past, the "count" of search results was simply the number of visible threads (i.e. `threads` in `rootThread` with `visible`: `true`). That meant that when a user force-expanded threads, the result count would appear to change.

It's a simple calculation to subtract the number of forced-visible threads from the total number of visible threads to get an accurate result count, but only if there's no duplication between the two sets (set A: `threads` with `visible: true`, set B: `thread`s that have been set to forced-visible).

A robust solution could be had by actually comparing items in `forcedVisible` when counting visible threads, but that was more code and more calculation, so I took this elementary approach for now.